### ISSUE TITLE
Double quotes escape bug

### ISSUE THREAD

**Issue by Nialit** [orientechnologies/orientdb#2690]
If I escape double quotes in SQL
```sql
update #13:0 add msg = {"":{"M":"\""}}
```
next msg is added(from V)
```json
{"M":"\\\""}
```
so that backslash is shielded too. Without backslashing I found no way to pass quotes. With backslashing extra backshash appears.

So,  one extra backslash is added.
If you replace the wrong value inside studio edit by right one
```json
{"msg":"\""}
```
then everything works fine and you get correct escaped double quotes. 

It's very annoing as needs extra server work when saving text, that contain double quotes.

**Comment by AaronSchulz** [comment]
I tried reproducing this and didn't get double escaping...but I can't get the full value to be added. Adding ```{"M":"\""}``` results in ```\"``` being added, rather than the full key/value tuple.

**Comment by Nialit** [comment]
I tried it with the latest release now "2.0-rc2" and the bug is remaining. I add a new value into embedded list property of the class, may be you make something different?

**Comment by AaronSchulz** [comment]
Yeah, it should be ```"```, not ```\"```, so that bug is indeed still there. I got so used to typing the backslash that I didn't even think about it.

The other problem I had was that if msg starts off as [] and I add ```{"":{"M":"\""}}```, then msg[0] is just the inner value instead of {M:value}. I'd expect msg[0]["M"] to have that value instead.

**Comment by luigidellaquila** [comment]
Hi guys, 

I just fixed #5132 in both 2.1.x and develop branches.
These two issues are strongly related, could you please check if it also solves your case?

Thanks

Luigi

**Comment by lvca** [comment]
Fixed because no answer.
